modules/tensor_mechanics/src/materials/ComputeEigenstrainBase.C

```diff
@@ -48,7 +49,11 @@ ComputeEigenstrainBase::initQpStatefulProperties()
 void
 ComputeEigenstrainBase::computeQpProperties()
 {
-  computeQpEigenstrain();
+  if (_t_step >= 1)
+    _step_zero = false;
+
+  if (!_step_zero)
```

Comment:
My only suggestion here is that we should put a comment here that explains why it is important to skip the first step.  How about this:
//Skip the eigenstrain calculation in step zero because no solution is done on that step, so computing
//it would result in an incorrect calculation of mechanical_strain, which is stateful.